service/storage_proxy.cc

```diff
@@ -2517,18 +2517,33 @@ storage_proxy::mutate_atomically(std::vector<mutation> mutations, db::consistenc
             m.partition().apply_delete(*schema, clustering_key_prefix::make_empty(), tombstone(now, gc_clock::now()));
 
             tracing::trace(_trace_state, "Sending a batchlog remove mutation");
-            return send_batchlog_mutation(std::move(m), db::consistency_level::ANY).handle_exception([] (std::exception_ptr eptr) {
-                slogger.error("Failed to remove mutations from batchlog: {}", eptr);
+            return send_batchlog_mutation(std::move(m), db::consistency_level::ANY).then_wrapped([] (future<result<>> f) {
+                auto print_exception = [] (const auto& ex) {
+                    slogger.error("Failed to remove mutations from batchlog: {}", ex);
+                };
+                if (f.failed()) {
+                    print_exception(f.get_exception());
+                } else if (result<> res = f.get(); res.has_error()) {
+                    res.assume_error().accept(print_exception);
```

Comment:
Can we add `operator<<` to `coordinator_error`?

Reply:
Yes, sounds like a good idea. I'll use it here and in the `operator<<` for `result_message::exception` (it already has the logic for printing the exception).